### ISSUE TITLE
Ability to pass `disallowTypeAhead` to RAC GridList

### ISSUE THREAD

**Issue by jaknas** [adobe/react-spectrum#4957]
### Provide a general summary of the feature here

I'd like to disable typeahead in GridList RAC component, due to some accessibility concerns ([speech input](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html), along with screen readers and "keyboard users who are prone to accidentally hit keys.").

### 🤔 Expected Behavior?

Passing `disallowTypeAhead` should disable typeahead feature

### 😯 Current Behavior

passing `disallowTypeAhead` does nothing

### 💁 Possible Solution

Have GridList accept `disallowTypeAhead` and have it pass further to `useSelectableList`.

### 🔦 Context

There have been some worries that having focus changes after clicking 1 character could disorient the user. For us, this risk outweighs the benefit of typeahead

(note: It'd be nice to be able to decide the threshold of keys after which focus should be moved, we could e.g increase it to two keys so accidental key presses won't move focus)

**Comment by reidbarber** [comment]
The argument could be made that disabling this would create a less accessible experience, since keyboard users may use this to more quickly navigate collections (and may be accustomed to it). I'm not saying there isn't a use case for allowing this to be disabled, but I think I would be hesitant to disable this.

**Comment by snowystinger** [comment]
Hey! Thanks for the issue. I've talked a bit with our accessibility team about this and we need more information about the specifics where this was reported. Could we see the actual report?

Our team points out that WCAG explicitly allows single key shortcuts when a component has focus and we believe that applies here.

We do think there may be some cases where disabling it would make sense, but we haven't defined it well, it's more of a feeling. For example, if there are only 2 entries typeahead is probably more confusing than simply using arrow keys.

That said, we do not think it should be disabled by default.

Thanks @jnurthen for having a look, tagging you so you get updates.

**Comment by ritz078** [comment]
If there are text input elements inside the rows, the focus shifts due to this while typing inside the input. I think this option should be disabled if an input field is focused.

**Comment by snowystinger** [comment]
closing as mentioned in previous comment that this will be solved via edit mode
